app/main.py

```diff
@@ -1 +1,38 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100,
+                 hidden: bool = False) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
```

Comment:
Code Efficiency: It's better to use the class name to access class attributes rather than `self`. So, in this case, you should use `Animal.alive` instead of `self.alive`. This is because `alive` is a class attribute and not an instance attribute. Here is the corrected version of your code:

python
class Animal:
    alive = []

    def __init__(self, name: str, health: int = 100, hidden: bool = False) -> None:
        self.name = name
        self.health = health
        self.hidden = hidden
        Animal.add_alive(self)
```

This will make your code more efficient and easier to understand.